### ISSUE TITLE
Can't apply/remove coupon code

### ISSUE THREAD

**Issue by zavan** [spree/spree#11014]
## Context
When I try to apply or remove a coupon code from the cart I get a 404 error.

## Expected Behavior
The coupon is applied/removed.

## Actual Behavior
A 404 error shows in the console due to a malformed URL:

`PATCH https://mydomain.com/store//store/api/v2/storefront/cart/apply_coupon_code?currency=USD&locale=en 404 (Not Found)`

Notice how the mount path is duplicated.

## Possible Fix
Looks like the problem is related to #10845. When I override the URL variables in my project to use `pathFor` instead of `localizedPathFor` it works:

```js
Spree.routes.api_v2_storefront_cart_apply_coupon_code = Spree.pathFor('api/v2/storefront/cart/apply_coupon_code')

Spree.routes.api_v2_storefront_cart_remove_coupon_code = function(couponCode) {
  return Spree.pathFor('api/v2/storefront/cart/remove_coupon_code/' + couponCode)
}
```

Not sure why but that's how the one to apply the coupon is defined in https://github.com/spree/spree/blob/v4.2.3.1/api/app/assets/javascripts/spree/api/main.js#L14

It's later overridden (I believe) in https://github.com/spree/spree/blob/v4.2.3.1/frontend/app/assets/javascripts/spree/frontend.js#L47

## Steps to Reproduce

1. Create a promotion with a code
2. Add a product to the cart
3. Try to apply the coupon to the cart

## Your Environment

* Version used: 4.2.3.1

**Comment by zavan** [comment]
The same problem is happening [here](https://github.com/spree/spree/blob/v4.2.3.1/frontend/app/assets/javascripts/spree/frontend/checkout/address.js#L11) when loading the countries for the checkout address. 

In that case I had to override the whole file to change it to use `pathFor` instead of `localizedPathFor` for it to work.

**Comment by zavan** [comment]
This has been fixed in [v4.2.5](https://github.com/spree/spree/releases/tag/v4.2.5) with https://github.com/spree/spree/commit/432a890871d38498486ab3c0ea50b8bd490b1d31.
